src/gordon_gcp/plugins/service/gdns_publisher.py

```diff
@@ -48,6 +50,116 @@
 DNS_CHANGES_DONE = 'done'
 
 
+class GDNSPublisherBuilder:
+    """Build and configure a :class:`GDNSPublisher` object.
+
+    Args:
+        config (dict): Google Cloud DNS API related configuration.
+        success_channel (asyncio.Queue): Queue to place a successfully
+            published message to be further handled by the ``gordon``
+            core system.
+        error_channel (asyncio.Queue): Queue to place a message met
+            with errors to be further handled by the ``gordon`` core
+            system.
+        kwargs (dict): Additional keyword arguments to pass to the
+            publisher.
+    """
+    def __init__(self, config, success_channel, error_channel, **kwargs):
+        self.config = config
+        self.success_channel = success_channel
+        self.error_channel = error_channel
+        self.kwargs = kwargs
+        self.validate_config_funcs = [
+            self._validate_keyfile, self._validate_project,
+            self._validate_dns_zone, self._validate_managed_zone,
+            self._validate_publish_timeout, self._validate_default_ttl
+        ]
+
+    def _validate_keyfile(self, errors):
+        # TODO (lynn): keyfile won't be required once we support other
+        #              auth methods
+        if not self.config.get('keyfile'):
+            msg = ('The path to a Service Account JSON keyfile is required to '
+                   'authenticate for Google Cloud DNS.')
+            errors.append(msg)
+
+    def _validate_project(self, errors):
+        if not self.config.get('project'):
+            msg = 'The GCP project where Cloud DNS is located is required.'
+            errors.append(msg)
+
+    def _validate_dns_zone(self, errors):
+        if not self.config.get('dns_zone'):
+            msg = 'A dns zone is required to build correct A records.'
+            errors.append(msg)
+        elif not self.config.get('dns_zone', '').endswith('.'):
+            msg = ('A dns zone must be an FQDN and end with the root zone '
+                   '(".").')
+            errors.append(msg)
+
+    def _validate_managed_zone(self, errors):
+        if not self.config.get('managed_zone'):
+            msg = ('A managed zone is required to publish records to Google '
+                   'Cloud DNS.')
+            errors.append(msg)
+
+    def _validate_publish_timeout(self, errors):
+        try:
+            timeout = float(self.config.get('publish_wait_timeout', 60))
```

Comment:
My comment wasn't so much about float v int, but that instead casting and type errors to infer numericalness, we could do something like
```python
if 'publish_wait_timeout' in self.config:
    timeout = self.config.get('publish_wait_timeout')
    if not isinstance(timeout, numbers.Number) or timeout < 0:
        msg =  '"publish_wait_timeout" must be a positive number."
        errors.append(msg)
```
It would have the added benefit of not supporting number-like strings, like `publish_wait_timeout = "3"`, which I think is a good thing to avoid.